src/Diagram/DiagramMemento.java

```diff
@@ -13,21 +13,21 @@ public class DiagramMemento {
     private HashMap<String, Class> classList;
     private HashMap<String, Relationship> relationshipList;
 
-    public DiagramMemento(Diagram diagram, final String title) {
-
-        if (title == null) {
-            throw new NullPointerException("invalid param in Diagram constructor");
-        }
+    public DiagramMemento(Diagram diagram) {
         this.diagram = diagram;
-        this.title = title;
-        this.classList = new HashMap<>();
-        this.relationshipList = new HashMap<>();
+        this.title = diagram.getTitle();
+        this.classList = new HashMap<>(diagram.getClassList());
+        this.relationshipList = new HashMap<>(diagram.getRelationshipList());
     }
 
-    public void restore() {
-        this.diagram.setTitle(this.title);
-        this.diagram.setSaveLocation(saveLocation);
-        this.diagram.setClassList(this.classList);
-        this.diagram.setRelationshipList(relationshipList);
-    }
+    public Diagram getDiagram() {return this.diagram;}
```

Comment:
This method right here just needs one test!

Reply:
Test for this has been added!